src/App.tsx

```diff
@@ -1,24 +1,75 @@
-/* eslint-disable max-len */
-/* eslint-disable jsx-a11y/control-has-associated-label */
-import React from 'react';
-import { UserWarning } from './UserWarning';
+import React, {
+  useCallback,
+  useContext,
+  useEffect,
+  useState,
+} from 'react';
 
-const USER_ID = 0;
+import { getTodos } from './api/todos';
+import { UserWarning } from './UserWarning';
+import { TodoList } from './components/TodoList';
+import { ErrorMessage } from './components/ErrorMessage';
+import { Loader } from './components/Loader';
+import { AppContext } from './components/AppContext';
+import { Header } from './components/Header';
+import { Footer } from './components/Footer';
 
 export const App: React.FC = () => {
-  if (!USER_ID) {
+  const [isLoadingTodos, setIsLoadingTodos] = useState(false);
+
+  // eslint-disable-next-line no-unused-vars
+  const {
+    userId,
+    allTodos,
+    setAllTodos,
+    setShouldShowError,
+    showError,
+  } = useContext(AppContext);
+
+  const loadTodosFromServer = useCallback(async () => {
+    setIsLoadingTodos(true);
+    setShouldShowError(false);
+
+    try {
+      const todosFromServer = await getTodos(userId);
+
+      setAllTodos(todosFromServer);
+    } catch {
+      showError('Unable to load todos');
+    } finally {
+      setIsLoadingTodos(false);
+    }
+  }, []);
```

Comment:
The dependency array in the `useEffect` is missing the `loadTodosFromServer` dependency. Add it to the array to avoid potential issues. 😊